src/checkpoint/container/queue_serialize.h

```diff
@@ -0,0 +1,104 @@
+/*
+//@HEADER
+// *****************************************************************************
+//
+//                             queue_serialize.h
+//                           DARMA Toolkit v. 1.0.0
+//                 DARMA/checkpoint => Serialization Library
+//
+// Copyright 2020 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS). Under the terms of Contract DE-NA0003525 with NTESS, the U.S.
+// Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact darma@sandia.gov
+//
+// *****************************************************************************
+//@HEADER
+*/
+
+#if !defined INCLUDED_CHECKPOINT_CONTAINER_QUEUE_SERIALIZE_H
+#define INCLUDED_CHECKPOINT_CONTAINER_QUEUE_SERIALIZE_H
+
+#include "checkpoint/common.h"
+
+#include <queue>
+
+namespace checkpoint {
+
+template <typename Serializer, typename T>
+void serialize(Serializer& s, std::queue<T> q) {
+  serializeQueue(s, q);
+}
+
+template <
+  typename SerializerT,
+  typename T,
+  typename = std::enable_if_t<
+    std::is_same<
+      SerializerT,
+      checkpoint::Footprinter
+    >::value
+  >
+>
+void serializeQueue(SerializerT& s, std::queue<T> q) {
```

Comment:
Per discussion on today's call, we prefer to just count `q.size() * sizeof(T)`, and sacrifice some accuracy to avoid the potential pitfall of copying and iterating over every element.

Reply:
That would shift us back to taking the queue arguments by `const&`

Reply:
done, the inaccuracy will be bigger when the underlying container is `std::vector` (default for priority queue), but there's no access to capacity in that case

Reply:
I realize that for the specific cases in the runtime of these containers, we can handle footprinting specially, by keeping a running measure of the footprint of the contents, to add to the footprint explicitly.

Which means, the footprint interface needs to be able to accept "add arbitrary byte count"